distributed/tests/test_scheduler.py

```diff
@@ -717,6 +717,15 @@ def test_retire_workers(c, s, a, b):
     assert not workers
 
 
+@gen_cluster(client=True, timeout=1000)
+def test_retire_workers_no_suspicious_tasks(c, s, a, b):
+    c.submit(slowinc, 100, delay=0.5, workers=a.address, allow_other_workers=True)
```

Comment:
You will need to hold on to this future.  Otherwise the task will be garbage-collected.

Reply:
If flake8 complains at you then I recommend verifying that it actually computes after you retire `a` by yielding on it and checking the value.